### PR TITLE
fix compile error

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
fix compile error

chip/efm32_leserial.c: In function 'efm32_restoreuartint': Error: chip/efm32_leserial.c:329:37: error: 'len' undeclared (first use in this function); did you mean 'ien'?
  329 |   efm32_restoreuartint_nolock(priv, len);
      |                                     ^~~
      |                                     ien

## Impact
efm32_leserial.c
## Testing
ci

### CODE REVIEW

**Review by lupyuen — APPROVED** [review]

File: arch/arm/src/gd32f4/gd32f4xx_serial.c [hunk 1]
```diff
@@ -159,6 +158,7 @@ struct up_dev_s
   const uint8_t   stop_2bits;    /* True: Configure with 2 stop bits instead of 1 */
   const uint32_t  tx_gpio;       /* USART TX GPIO pin configuration */
   const uint32_t  rx_gpio;       /* USART RX GPIO pin configuration */
+  spinlock_t      lock;          /* Spinlock */
```
Comment (hartmannathan) [inline arch/arm/src/gd32f4/gd32f4xx_serial.c:161]: Is this change intended to be included in same PR?

Comment (hujun260) [reply arch/arm/src/gd32f4/gd32f4xx_serial.c:161]: yes， It also resulted in compilation errors.
Configuration/Tool: gd32f470zk-eval/nsh,CONFIG_ARM_TOOLCHAIN_GNU_EABI
2024-12-27 02:06:08
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Building NuttX...
Error: chip/gd32f4xx_serial.c:441:4: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
  441 |   .lock          = SP_UNLOCKED,
      |    ^~~~
      |    clock
In file included from chip/gd32f4xx_serial.c:42:
chip/gd32f4xx_serial.c: In function 'up_disableusartint':
Error: chip/gd32f4xx_serial.c:1139:36: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1139 |   flags = spin_lock_irqsave(&priv->lock);
      |                                    ^~~~
Error: chip/gd32f4xx_serial.c:1173:33: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1173 |   spin_unlock_irqrestore(&priv->lock, flags);
      |                                 ^~~~
chip/gd32f4xx_serial.c: In function 'up_restoreusartint':
Error: chip/gd32f4xx_serial.c:1184:36: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1184 |   flags = spin_lock_irqsave(&priv->lock);
      |                                    ^~~~
Error: chip/gd32f4xx_serial.c:1188:33: error: 'struct up_dev_s' has no member named 'lock'; did you mean 'clock'?
 1188 |   spin_unlock_irqrestore(&priv->lock, flags);
      |                                 ^~~~

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements. While it identifies a compile error fix, it lacks crucial information.

Here's what's missing:

* **Summary:** Needs more detail. *Why* was `len` undeclared?  What was the root cause of the compile error? What exactly was changed to fix it?  Just stating "fix compile error" is insufficient.  Mention the affected functionality (e.g., serial driver).
* **Impact:**  Saying `efm32_leserial.c` is affected is not enough. Answer all the impact questions specifically.  Does this fix introduce any user-facing changes?  Does it affect any particular boards?  Does it require documentation updates?  Most will likely be "NO", but you still need to state that explicitly.
* **Testing:** "ci" is not adequate. Provide actual build and test logs *before* and *after* the change, demonstrating that the issue is resolved and no new issues are introduced. Specify the host and target environments used for testing (OS, architecture, board, configuration). Ideally,  show that the serial driver functions correctly after the fix.

**Example of a better summary and testing section:**

**Summary:**

This PR fixes a compile error in `chip/efm32_leserial.c` caused by the undeclared variable `len` within the `efm32_restoreuartint` function. This error occurred because `len` was used without being declared, likely intended to be the interrupt enable state.  The fix introduces a new variable `uart_intstate` to store the interrupt enable state and uses it correctly within `efm32_restoreuartint` and `efm32_restoreuartint_nolock`.  This resolves the compilation issue and restores the correct handling of serial port interrupts.

**Impact:**

* Is new feature added? NO
* Is existing feature changed? NO (bug fix)
* Impact on user: NO
* Impact on build: NO (fixes a broken build)
* Impact on hardware: NO
* Impact on documentation: NO
* Impact on security: NO
* Impact on compatibility: NO
* Anything else to consider? NO


**Testing:**

**Build Host:** Linux, x86_64, GCC 9.4.0
**Target:** QEMU, ARM Cortex-M3, EFM32 Giant Gecko (sim)

**Testing logs before change:**

```
chip/efm32_leserial.c: In function 'efm32_restoreuartint':
chip/efm32_leserial.c:329:37: error: 'len' undeclared (first use in this function); did you mean 'ien'?
  329 |   efm32_restoreuartint_nolock(priv, len);
      |                                     ^~~
      |                                     ien
make: *** [<builtin>: chip/efm32_leserial.o] Error 1
```

**Testing logs after change:**

```
... (build output showing successful compilation) ...

nsh> help  # Verify serial console works
... (nsh help output)
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@jerpelea this patch need cherry-pick to release branch too.
